src/App/Platforms/Android/Services/DeviceActionService.cs

```diff
@@ -553,6 +558,107 @@ public long GetActiveTime()
             // ref: https://developer.android.com/reference/android/os/SystemClock#elapsedRealtime()
             return SystemClock.ElapsedRealtime();
         }
+
+        public async Task ReturnToPasskeyAfterUnlockAsync()
+        {
+            var activity = Microsoft.Maui.ApplicationModel.Platform.CurrentActivity as MainActivity;
+            if (activity == null)
+            {
+                return;
+            }
+
+            try
+            {
+                var request = AndroidX.Credentials.Provider.PendingIntentHandler.RetrieveBeginGetCredentialRequest(activity.Intent);
+                var response = new AndroidX.Credentials.Provider.BeginGetCredentialResponse();;
+                List<AndroidX.Credentials.Provider.CredentialEntry> credentialEntries = null;
+                foreach (var option in request.BeginGetCredentialOptions)
+                {
+                    var credentialOption = option as AndroidX.Credentials.Provider.BeginGetPublicKeyCredentialOption;
+                    if (credentialOption != null)
+                    {
+                        credentialEntries ??= new List<AndroidX.Credentials.Provider.CredentialEntry>();
+                        credentialEntries.AddRange(await PopulatePasskeyDataAsync(request.CallingAppInfo, credentialOption));
+                    }
+                }
+
+                if (credentialEntries != null)
+                {
+                    response = new AndroidX.Credentials.Provider.BeginGetCredentialResponse.Builder()
+                        .SetCredentialEntries(credentialEntries)
+                        .Build();
+                }
+
+                var result = new Android.Content.Intent();
+                AndroidX.Credentials.Provider.PendingIntentHandler.SetBeginGetCredentialResponse(result, response);
+
+                activity.SetResult(Result.Ok, result);
+                activity.Finish();
+            }
+            catch (Exception ex)
+            {
+                Bit.Core.Services.LoggerHelper.LogEvenIfCantBeResolved(ex);
+
+                activity.SetResult(Result.Canceled);
+                activity.Finish();
+            }
+        }
+
+        #region CODE_THAT_NEEDS_TO_BE_MOVED_ELSEWHERE
+        //TODO: This region needs to be moved somewhere else where it can be shared with the code in CredentialProviderService.cs
+        private async Task<List<AndroidX.Credentials.Provider.CredentialEntry>> PopulatePasskeyDataAsync(AndroidX.Credentials.Provider.CallingAppInfo callingAppInfo,
```

Comment:
thought: we could consider exposing and using `Authenticator.SilentCredentialDiscoveryAsync` to implement this

Reply:
I have moved the "duplicated" code into CredentialHelpers.cs
But if you think it makes sense to change that code to call that we could do it anyway.